plugin/src/main/java/com/spotify/plugin/dockerfile/BuildMojo.java

```diff
@@ -145,6 +156,15 @@ static String buildImage(@Nonnull DockerClient dockerClient,
       buildParameters.add(DockerClient.BuildParam.noCache());
     }
 
+    if (buildArgs != null && !buildArgs.isEmpty()) {
+      try {
+        final String encodedBuildArgs = URLEncoder.encode(new Gson().toJson(buildArgs), "utf-8");
```

Comment:
I could be wrong but I think Jersey (used by docker-client) encode the name/value of the BuildParam as query params before they are passed to the Docker Remote API. Are you sure the encoding is needed here too?